File: pages/settings_page.py
import sys, os
sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

import util.theme as _theme
import config as _cfg

from PySide6.QtWidgets import (
    QWidget, QHBoxLayout, QVBoxLayout, QStackedWidget,
    QLabel, QPushButton, QFrame, QSizePolicy, QScrollArea,
)
from PySide6.QtCore import Qt, QThread, Signal

from pages import BasePage, BaseSetting
from util.widgets import ThemedComboBox, ThemedToggle
from util.log_util import get_tagged_logger

logger = get_tagged_logger("设置", __name__)


def _C() -> dict:
    return _theme.get()


def build_setting_qss() -> str:
    C = _C()
    return f"""
#SettingsSidebar {{
    background: {C["sidebar"]};
    border-bottom: 1px solid {C["border"]};
}}
#SettingNavBtn {{
    background: transparent;
    border: none;
    border-bottom: 2px solid transparent;
    padding: 0 16px;
    color: {C["text_muted"]};
    font-size: 13px;
}}
#SettingNavBtn:hover {{
    background: {C["hover"]};
    color: {C["text"]};
    border-bottom: 2px solid transparent;
}}
#SettingNavBtn[active=true] {{
    background: transparent;
    color: {C["text"]};
    font-weight: 600;
    border-bottom: 2px solid {C["active_line"]};
}}
#SettingContent  {{ background: {C["bg"]}; }}
#SettingCard     {{ background: {C["card"]}; border-radius: 10px;
                    border: 1px solid {C["border"]}; }}
#SettingCardTitle   {{ font-size: 13px; font-weight: 600; color: {C["text_muted"]}; }}
#SettingPageTitle   {{ font-size: 20px; font-weight: 600; color: {C["text"]}; }}
QScrollBar:vertical {{ background: transparent; width: 4px; }}
QScrollBar::handle:vertical {{ background: {C["border"]}; border-radius: 2px; min-height: 20px; }}
QScrollBar::add-line:vertical, QScrollBar::sub-line:vertical {{ height: 0; }}
"""


# ─────────────────────────────────────────────
# 顶部 Tab 按钮
# ─────────────────────────────────────────────
class SettingNavBtn(QPushButton):
    H = 46

    def __init__(self, label: str, parent=None):
        super().__init__(label, parent)
        self.setObjectName("SettingNavBtn")
        self.setFixedHeight(self.H)
        self.setSizePolicy(QSizePolicy.Fixed, QSizePolicy.Fixed)
        self.setCursor(Qt.PointingHandCursor)

    def set_active(self, val: bool):
        self.setProperty("active", val)
        self.style().unpolish(self)
        self.style().polish(self)


# ─────────────────────────────────────────────
# 内置面板：系统
# ─────────────────────────────────────────────
class SystemSettings(BaseSetting):
    name  = "系统"
    order = 0

    def __init__(self):
        super().__init__()
        lay = QVBoxLayout(self)
        lay.setContentsMargins(28, 24, 28, 24)
        lay.setSpacing(16)

        t = QLabel("系统")
        t.setObjectName("SettingPageTitle")
        lay.addWidget(t)

        card, inner = self.build_section("颜色主题")

        row = QHBoxLayout()
        row.setSpacing(12)

        lbl = QLabel("系统颜色主题")
        lbl.setStyleSheet("background: transparent; font-size: 14px;")
        row.addWidget(lbl)
        row.addStretch()

        self._combo = ThemedComboBox()
        self._combo.addItems(_theme.names())
        self._combo.setCurrentText(_theme.current_name())
        self._combo.setFixedHeight(34)
        self._combo.setMinimumWidth(160)
        self._combo.currentTextChanged.connect(_theme.set_theme)
        row.addWidget(self._combo)

        inner.addLayout(row)
        lay.addWidget(card)

        card2, inner2 = self.build_section("行为")

        row2 = QHBoxLayout()
        row2.setSpacing(12)
        lbl2 = QLabel("关闭时缩小到托盘")
        lbl2.setStyleSheet("background: transparent; font-size: 14px;")
        row2.addWidget(lbl2)
        row2.addStretch()

        self._tray_toggle = ThemedToggle("minimize_to_tray", default=True)
        self._tray_toggle.toggled.connect(self._on_tray_setting_changed)
        row2.addWidget(self._tray_toggle)

        inner2.addLayout(row2)
        lay.addWidget(card2)

        lay.addStretch()

    def _on_tray_setting_changed(self, enabled: bool) -> None:
        root = self.window()
        if hasattr(root, "apply_minimize_to_tray_setting"):
            root.apply_minimize_to_tray_setting(enabled)



# ─────────────────────────────────────────────
# 内置面板：账号
# ─────────────────────────────────────────────
class _AccountLoginThread(QThread):
    finished = Signal(bool, str)

    def run(self):
        try:
            from listener.login import do_login
            ok = do_login()
            self.finished.emit(ok, "" if ok else "登录未完成或已取消")
        except Exception as e:
            logger.error("登录线程异常: %s", e)
            self.finished.emit(False, str(e))


class AccountSettings(BaseSetting):
    name  = "账号"
    order = 1

    def __init__(self):
        super().__init__()
        lay = QVBoxLayout(self)
        lay.setContentsMargins(28, 24, 28, 24)
        lay.setSpacing(16)

        t = QLabel("账号")
        t.setObjectName("SettingPageTitle")
        lay.addWidget(t)

        card, inner = self.build_section("登录状态")
        self._status = QLabel("检测中...")
        self._status.setObjectName("PageSubtitle")
        inner.addWidget(self._status)

        self._btn = QPushButton("重新登录")
        self._btn.setFixedHeight(34)
        self._btn.setCursor(Qt.PointingHandCursor)
        self._btn.setStyleSheet(f"""
            QPushButton {{
                background: {_C()["active_line"]}; color: #fff;
                border: none; border-radius: 6px;
                font-size: 13px; padding: 0 16px;
            }}
            QPushButton:hover {{ background: #7B9FFF; }}
        """)
        self._btn.clicked.connect(self._do_login)
        inner.addWidget(self._btn)
        lay.addWidget(card)
        lay.addStretch()
        self._login_thread: _AccountLoginThread | None = None

        self._refresh()

    def _refresh(self):
        from listener.login import get_login_ui_state
        text, _ = get_login_ui_state()
        self._status.setText(text)

    def _reset_login_btn(self) -> None:
        self._btn.setText("重新登录")
        self._btn.setEnabled(True)

    def _do_login(self):
        if self._login_thread and self._login_thread.isRunning():
            return
        self._btn.setText("登录中...")
        self._btn.setEnabled(False)
        try:
            self._login_thread = _AccountLoginThread(self)
            self._login_thread.finished.connect(self._on_login_done)
            self._login_thread.start()
        except Exception as e:
            logger.error("登录线程启动失败: %s", e)
            self._reset_login_btn()
            self._status.setText(f"登录失败: {e}")

    def _on_login_done(self, ok: bool, err: str):
        self._reset_login_btn()
        if ok:
            self._refresh()
        else:
            self._status.setText(f"登录失败: {err}" if err else "登录未完成或已取消")


# ─────────────────────────────────────────────
# SettingsPage 主体
# ─────────────────────────────────────────────
class SettingsPage(BasePage):
    PAGE_ICON = "⚙"
    PAGE_NAME = "设置"
    PAGE_SIZE = 5

    def __init__(self):
        super().__init__()
        self._nav_btns: list[SettingNavBtn] = []
        self._panels:   list[QWidget]       = []
        self._cur_page  = 0
        self._build()
        self._navigate(0)

        self.setStyleSheet(build_setting_qss())
        _theme.on_change(lambda _: self.setStyleSheet(build_setting_qss()))

    def _build(self):
        import pages as _pages

        lay = QVBoxLayout(self)
        lay.setContentsMargins(0, 0, 0, 0)
        lay.setSpacing(0)

        topbar = QWidget()
        topbar.setObjectName("SettingsSidebar")
        topbar.setFixedHeight(46)
        tb_lay = QHBoxLayout(topbar)
        tb_lay.setContentsMargins(8, 0, 8, 0)
        tb_lay.setSpacing(0)

        self._prev_btn = QPushButton("← 上一页")
        self._prev_btn.setObjectName("SettingNavBtn")
        self._prev_btn.setFixedHeight(46)
        self._prev_btn.setCursor(Qt.PointingHandCursor)
        self._prev_btn.clicked.connect(self._prev_page)
        tb_lay.addWidget(self._prev_btn)

        self._tab_container = QWidget()
        self._tab_layout = QHBoxLayout(self._tab_container)
        self._tab_layout.setContentsMargins(0, 0, 0, 0)
        self._tab_layout.setSpacing(0)
        tb_lay.addWidget(self._tab_container)

        tb_lay.addStretch()

        self._next_btn = QPushButton("下一页 →")
        self._next_btn.setObjectName("SettingNavBtn")
        self._next_btn.setFixedHeight(46)
        self._next_btn.setCursor(Qt.PointingHandCursor)
        self._next_btn.clicked.connect(self._next_page)
        tb_lay.addWidget(self._next_btn)

        lay.addWidget(topbar)

        scroll = QScrollArea()
        scroll.setWidgetResizable(True)
        scroll.setHorizontalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
        scroll.setObjectName("SettingContent")

        self._stack = QStackedWidget()
        self._stack.setObjectName("SettingContent")

        for cls in _pages.SETTINGS:
            panel = cls()
            btn   = SettingNavBtn(cls.name)
            btn.clicked.connect(
                lambda _, i=len(self._nav_btns): self._navigate(i)
            )
            self._nav_btns.append(btn)
            self._panels.append(panel)
            self._stack.addWidget(panel)

        scroll.setWidget(self._stack)
        lay.addWidget(scroll)

        self._update_page()

    def _total_pages(self) -> int:
        import math
        return max(1, math.ceil(len(self._nav_btns) / self.PAGE_SIZE))

    def _update_page(self):
        while self._tab_layout.count():
            item = self._tab_layout.takeAt(0)
            if item.widget():
                item.widget().setParent(None)

        start = self._cur_page * self.PAGE_SIZE
        for btn in self._nav_btns[start : start + self.PAGE_SIZE]:
            self._tab_layout.addWidget(btn)
            btn.show()

        has_prev = self._cur_page > 0
        has_next = self._cur_page < self._total_pages() - 1

        self._prev_btn.setEnabled(has_prev)
        self._next_btn.setEnabled(has_next)

        C = _C()
        for btn, has in [(self._prev_btn, has_prev), (self._next_btn, has_next)]:
            btn.setStyleSheet(
                f"QPushButton {{ background: transparent; border: none;"
                f" border-bottom: 2px solid transparent; padding: 0 10px;"
                f" font-size: 13px;"
                f" color: {C['text'] if has else C['text_muted']}; }}"
                f"QPushButton:hover {{ background: {C['hover'] if has else 'transparent'}; }}"
            )

    def _prev_page(self):
        if self._cur_page > 0:
            self._cur_page -= 1
            self._update_page()

    def _next_page(self):
        if self._cur_page < self._total_pages() - 1:
            self._cur_page += 1
            self._update_page()

    def _navigate(self, index: int):
        self._stack.setCurrentIndex(index)
        for i, btn in enumerate(self._nav_btns):
            btn.set_active(i == index)

File: pages/settings_page.cpp
// pages/settings_page.cpp — 设置页，对齐旧 PySide settings_page.py。
// 顶部二级导航（含上一页/下一页分页）+ 系统 / 账号 / 工具设置三块面板。

namespace liveaio::pages {

using liveaio::util::deferNextTick;

// 旧 BaseSetting.build_section
struct SettingSection {
    QFrame* card = nullptr;
    QVBoxLayout* body = nullptr;
};

static SettingSection settingSection(const QString& title, QWidget* parent) {
    SettingSection out;
    out.card = new QFrame(parent);
    out.card->setObjectName(QStringLiteral("SettingCard"));
    out.body = new QVBoxLayout(out.card);
    out.body->setContentsMargins(20, 16, 20, 16);
    out.body->setSpacing(12);
    auto* lbl = new QLabel(title, out.card);
    lbl->setObjectName(QStringLiteral("SettingCardTitle"));
    out.body->addWidget(lbl);
    return out;
}

class SettingNavBtn final : public QPushButton {
public:
    explicit SettingNavBtn(const QString& label, QWidget* parent = nullptr)
        : QPushButton(label, parent) {
        setObjectName(QStringLiteral("SettingNavBtn"));
        setFixedHeight(46);
        setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed);
        setCursor(Qt::PointingHandCursor);
    }

    void setActive(bool on) {
        setProperty("active", on);
        style()->unpolish(this);
        style()->polish(this);
    }
};

class SettingPanel : public QWidget {
public:
    explicit SettingPanel(QWidget* parent = nullptr) : QWidget(parent) {}
    virtual QString panelName() const = 0;
    virtual void onCorePacket(const QJsonObject&) {}
    virtual void refreshTheme() {}
};

// ── 系统 ──────────────────────────────────────
class SystemSettingsPanel final : public SettingPanel {
public:
    explicit SystemSettingsPanel(CoreClient* core, QWidget* parent = nullptr)
        : SettingPanel(parent), core_(core) {
        auto* lay = new QVBoxLayout(this);
        lay->setContentsMargins(28, 24, 28, 24);
        lay->setSpacing(16);

        auto* title = new QLabel(QStringLiteral("系统"), this);
        title->setObjectName(QStringLiteral("SettingPageTitle"));
        lay->addWidget(title);

        SettingSection themeCard = settingSection(QStringLiteral("颜色主题"), this);
        auto* row = new QHBoxLayout;
        row->setSpacing(12);
        auto* lbl = new QLabel(QStringLiteral("系统颜色主题"), themeCard.card);
        lbl->setStyleSheet(QStringLiteral("background: transparent; font-size: 14px;"));
        row->addWidget(lbl);
        row->addStretch();
        combo_ = new ThemedComboBox(themeCard.card);
        combo_->addItems(liveaio::util::themeNames());
        combo_->setCurrentText(liveaio::util::currentThemeName());
        combo_->setFixedHeight(34);
        combo_->setMinimumWidth(160);
        combo_->setOnChange([](const QString& name) { liveaio::util::setTheme(name); });
        row->addWidget(combo_);
        themeCard.body->addLayout(row);
        lay->addWidget(themeCard.card);

        SettingSection behaviorCard = settingSection(QStringLiteral("行为"), this);
        auto* row2 = new QHBoxLayout;
        row2->setSpacing(12);
        auto* lbl2 = new QLabel(QStringLiteral("关闭后缩小到托盘"), behaviorCard.card);
        lbl2->setStyleSheet(QStringLiteral("background: transparent; font-size: 14px;"));
        row2->addWidget(lbl2);
        row2->addStretch();
        trayToggle_ = new ThemedToggle(QStringLiteral("minimize_to_tray"), true, behaviorCard.card);
        g_minimizeToTray = trayToggle_->value();
        trayToggle_->setOnToggled([](bool on) { g_minimizeToTray = on; });
        row2->addWidget(trayToggle_);
        behaviorCard.body->addLayout(row2);
        lay->addWidget(behaviorCard.card);

        lay->addStretch();
    }

    QString panelName() const override { return QStringLiteral("系统"); }

    void onCorePacket(const QJsonObject& packet) override {
        const QString op = packet.value(QStringLiteral("op")).toString();
        if (op == QStringLiteral("config.value")) {
            const QJsonObject values = packet.value(QStringLiteral("values")).toObject();
            if (values.contains(QStringLiteral("theme"))) {
                applyThemeFromCore(values.value(QStringLiteral("theme")).toString());
            }
            if (values.contains(QStringLiteral("minimize_to_tray"))) {
                applyTrayFromCore(values.value(QStringLiteral("minimize_to_tray")).toBool(true));
            }
        } else if (op == QStringLiteral("config.ok")) {
            const QString key = packet.value(QStringLiteral("key")).toString();
            if (key == QStringLiteral("theme")) {
                applyThemeFromCore(packet.value(QStringLiteral("value")).toString());
            } else if (key == QStringLiteral("minimize_to_tray")) {
                applyTrayFromCore(packet.value(QStringLiteral("value")).toBool(true));
            }
        }
    }

    void refreshTheme() override {
        if (combo_) {
            combo_->refreshTheme();
            const QString name = liveaio::util::currentThemeName();
            if (combo_->currentText() != name) combo_->setCurrentText(name);
        }
    }

private:
    void applyThemeFromCore(const QString& raw) {
        const QString name = liveaio::util::normalizeThemeName(raw);
        if (name == liveaio::util::currentThemeName()) return;
        liveaio::util::applyThemeName(name);
        if (combo_) combo_->setCurrentText(name);
    }

    void applyTrayFromCore(bool enabled) {
        g_minimizeToTray = enabled;
        if (trayToggle_) trayToggle_->setValue(enabled, false);
    }

    CoreClient* core_ = nullptr;
    ThemedComboBox* combo_ = nullptr;
    ThemedToggle* trayToggle_ = nullptr;
};

// ── 账号 ──────────────────────────────────────
class AccountSettingsPanel final : public SettingPanel {
public:
    explicit AccountSettingsPanel(CoreClient* core, QWidget* parent = nullptr)
        : SettingPanel(parent), core_(core) {
        auto* lay = new QVBoxLayout(this);
        lay->setContentsMargins(28, 24, 28, 24);
        lay->setSpacing(16);

        auto* title = new QLabel(QStringLiteral("账号"), this);
        title->setObjectName(QStringLiteral("SettingPageTitle"));
        lay->addWidget(title);

        SettingSection card = settingSection(QStringLiteral("登录状态"), this);
        status_ = new QLabel(QStringLiteral("检测中..."), card.card);
        status_->setObjectName(QStringLiteral("PageSubtitle"));
        card.body->addWidget(status_);

        btn_ = new QPushButton(QStringLiteral("重新登录"), card.card);
        btn_->setFixedHeight(34);
        btn_->setCursor(Qt::PointingHandCursor);
        QObject::connect(btn_, &QPushButton::clicked, this, [this]() { startLogin(); });
        card.body->addWidget(btn_);
        lay->addWidget(card.card);
        lay->addStretch();

        refreshTheme();
    }

    QString panelName() const override { return QStringLiteral("账号"); }

    void onCorePacket(const QJsonObject& packet) override {
        if (packet.value(QStringLiteral("op")).toString() != QStringLiteral("login.state")) return;
        status_->setText(packet.value(QStringLiteral("text")).toString());
        btn_->setText(QStringLiteral("重新登录"));
        enabled_ = packet.value(QStringLiteral("can_login")).toBool(true);
        applyBtnStyle();
    }

    void queryLogin() {
        if (core_) core_->uiCommand(QStringLiteral("login.query"));
    }

    void refreshTheme() override { applyBtnStyle(); }

private:
    void applyBtnStyle() {
        btn_->setEnabled(enabled_);
        btn_->setStyleSheet(enabled_ ? qssSuccess(34) : qssDisabled(34));
    }

    void startLogin() {
        btn_->setText(QStringLiteral("登录中..."));
        enabled_ = false;
        applyBtnStyle();
        if (core_) core_->uiCommand(QStringLiteral("login.start"));
    }

    CoreClient* core_ = nullptr;
    QLabel* status_ = nullptr;
    QPushButton* btn_ = nullptr;
    bool enabled_ = true;
};

// ── 工具设置 ──────────────────────────────────
class ToolsSettingsPanel final : public SettingPanel {
public:
    explicit ToolsSettingsPanel(QWidget* parent = nullptr) : SettingPanel(parent) {
        auto* lay = new QVBoxLayout(this);
        lay->setContentsMargins(28, 24, 28, 24);
        lay->setSpacing(16);
        auto* title = new QLabel(QStringLiteral("工具设置"), this);
        title->setObjectName(QStringLiteral("SettingPageTitle"));
        lay->addWidget(title);

        SettingSection card = settingSection(QStringLiteral("备忘录"), this);
        card.body->addWidget(new QLabel(QStringLiteral("详细设置请在工具内调整"), card.card));
        lay->addWidget(card.card);
        lay->addStretch();
    }

    QString panelName() const override { return QStringLiteral("工具设置"); }
};

// ── SettingsPage ──────────────────────────────
class SettingsPage final : public BasePage {
public:
    static constexpr int kPageSize = 5;

    explicit SettingsPage(CoreClient* core, QWidget* parent = nullptr)
        : BasePage(parent), core_(core) {
        auto* lay = new QVBoxLayout(this);
        lay->setContentsMargins(0, 0, 0, 0);
        lay->setSpacing(0);

        auto* topbar = new QWidget(this);
        topbar->setObjectName(QStringLiteral("SettingsSidebar"));
        topbar->setFixedHeight(46);
        auto* tbLay = new QHBoxLayout(topbar);
        tbLay->setContentsMargins(8, 0, 8, 0);
        tbLay->setSpacing(0);

        prevBtn_ = new QPushButton(QStringLiteral("← 上一页"), topbar);
        prevBtn_->setObjectName(QStringLiteral("SettingNavBtn"));
        prevBtn_->setFixedHeight(46);
        prevBtn_->setCursor(Qt::PointingHandCursor);
        QObject::connect(prevBtn_, &QPushButton::clicked, this, [this]() {
            if (curPage_ > 0) {
                --curPage_;
                updatePage();
            }
        });
        tbLay->addWidget(prevBtn_);

        tabContainer_ = new QWidget(topbar);
        tabLayout_ = new QHBoxLayout(tabContainer_);
        tabLayout_->setContentsMargins(0, 0, 0, 0);
        tabLayout_->setSpacing(0);
        tbLay->addWidget(tabContainer_);
        tbLay->addStretch();

        nextBtn_ = new QPushButton(QStringLiteral("下一页 →"), topbar);
        nextBtn_->setObjectName(QStringLiteral("SettingNavBtn"));
        nextBtn_->setFixedHeight(46);
        nextBtn_->setCursor(Qt::PointingHandCursor);
        QObject::connect(nextBtn_, &QPushButton::clicked, this, [this]() {
            if (curPage_ < totalPages() - 1) {
                ++curPage_;
                updatePage();
            }
        });
        tbLay->addWidget(nextBtn_);
        lay->addWidget(topbar);

        auto* scroll = new QScrollArea(this);
        scroll->setWidgetResizable(true);
        scroll->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
        scroll->setObjectName(QStringLiteral("SettingContent"));

        stack_ = new QStackedWidget(scroll);
        stack_->setObjectName(QStringLiteral("SettingContent"));

        static const QStringList kPanelNames = {
            QStringLiteral("系统"), QStringLiteral("账号"), QStringLiteral("工具设置"),
        };
        panels_.resize(kPanelNames.size());
        panelFactories_ = {
            [this]() -> SettingPanel* { return new SystemSettingsPanel(core_, stack_); },
            [this]() -> SettingPanel* { return new AccountSettingsPanel(core_, stack_); },
            [this]() -> SettingPanel* { return new ToolsSettingsPanel(stack_); },
        };
        for (int i = 0; i < kPanelNames.size(); ++i) {
            auto* btn = new SettingNavBtn(kPanelNames.at(i), tabContainer_);
            QObject::connect(btn, &QPushButton::clicked, this, [this, i]() { navigate(i); });
            navBtns_.append(btn);
            auto* ph = new QWidget(stack_);
            ph->setObjectName(QStringLiteral("SettingPlaceholder"));
            panelPlaceholders_.append(ph);
            stack_->addWidget(ph);
        }

        scroll->setWidget(stack_);
        lay->addWidget(scroll);

        updatePage();
        navigate(0);
    }

    void onCorePacket(const QJsonObject& packet) override {
        for (SettingPanel* panel : panels_) {
            if (panel) panel->onCorePacket(packet);
        }
    }

    void refreshTheme() override {
        for (SettingPanel* panel : panels_) {
            if (panel) panel->refreshTheme();
        }
        updatePagerStyle();
    }

private:
    void ensurePanel(int index) {
        if (index < 0 || index >= panels_.size() || panels_[index]) return;
        if (index >= panelFactories_.size()) return;
        panels_[index] = panelFactories_[index]();
        QWidget* ph = panelPlaceholders_.value(index);
        if (!ph) return;
        const int idx = stack_->indexOf(ph);
        stack_->removeWidget(ph);
        ph->deleteLater();
        panelPlaceholders_[index] = nullptr;
        stack_->insertWidget(idx, panels_[index]);
    }
    int totalPages() const {
        return qMax(1, (navBtns_.size() + kPageSize - 1) / kPageSize);
    }

    void updatePage() {
        while (QLayoutItem* item = tabLayout_->takeAt(0)) {
            if (QWidget* w = item->widget()) w->setParent(tabContainer_);
            delete item;
        }
        for (auto* btn : navBtns_) btn->hide();
        const int start = curPage_ * kPageSize;
        for (int i = start; i < qMin(start + kPageSize, navBtns_.size()); ++i) {
            tabLayout_->addWidget(navBtns_[i]);
            navBtns_[i]->show();
        }
        prevBtn_->setEnabled(curPage_ > 0);
        nextBtn_->setEnabled(curPage_ < totalPages() - 1);
        updatePagerStyle();
    }

    void updatePagerStyle() {
        const auto& C = theme();
        const auto styleFor = [&C](bool active) {
            return QStringLiteral(
                "QPushButton { background: transparent; border: none;"
                " border-bottom: 2px solid transparent; padding: 0 10px; font-size: 13px;"
                " color: %1; }"
                "QPushButton:hover { background: %2; }"
            ).arg(active ? C.text : C.textMuted, active ? C.hover : QStringLiteral("transparent"));
        };
        prevBtn_->setStyleSheet(styleFor(prevBtn_->isEnabled()));
        nextBtn_->setStyleSheet(styleFor(nextBtn_->isEnabled()));
    }

    void navigate(int index) {
        ensurePanel(index);
        if (!panels_[index]) return;
        stack_->setCurrentIndex(stack_->indexOf(panels_[index]));
        for (int i = 0; i < navBtns_.size(); ++i) navBtns_[i]->setActive(i == index);
        if (panels_[index] && panels_[index]->panelName() == QStringLiteral("账号")) {
            if (auto* account = dynamic_cast<AccountSettingsPanel*>(panels_[index])) {
                account->queryLogin();
            }
        }
    }

    CoreClient* core_ = nullptr;
    QPushButton* prevBtn_ = nullptr;
    QPushButton* nextBtn_ = nullptr;
    QWidget* tabContainer_ = nullptr;
    QHBoxLayout* tabLayout_ = nullptr;
    QStackedWidget* stack_ = nullptr;
    QVector<SettingNavBtn*> navBtns_;
    QVector<SettingPanel*> panels_;
    QVector<QWidget*> panelPlaceholders_;
    QVector<std::function<SettingPanel*()>> panelFactories_;
    int curPage_ = 0;
};

}  // namespace liveaio::pages
